src/main/java/dansplugins/factionsystem/eventhandlers/DamageHandler.java

```diff
@@ -106,40 +106,77 @@ private void handleEntityDamage(Player attacker, EntityDamageByEntityEvent event
             logger.debug("Entity is an instance of a player. Returning.");
             return;
         }
-        Faction playersFaction = persistentData.getPlayersFaction(attacker.getUniqueId());
-        if (playersFaction == null) {
+
+        // If entity is protected, cancel the event and return
+        if (isEntityProtected(attacker, event.getEntity())) {
             event.setCancelled(true);
             return;
         }
 
-        if (isEntityProtected(event.getEntity())) {
-            cancelDamageIfNecessary(event, playersFaction);
-        }
+//        Faction playersFaction = persistentData.getPlayersFaction(attacker.getUniqueId());
```

Comment:
Prefer removing code over commenting it out - it is present in git history if it is needed in future and clutters the codebase when left in.

Reply:
Yeah let's remove any code that is currently commented out.

Reply:
done